modules/apps/101-interchain-swap/keeper/msg_server.go

```diff
@@ -18,20 +18,32 @@ func NewMsgServerImpl(keeper Keeper) types.MsgServer {
 
 var _ types.MsgServer = msgServer{}
 
-func (k Keeper) OnCreatePoolAcknowledged(ctx sdk.Context, msg *types.MsgCreatePoolRequest) {
+func (k Keeper) OnCreatePoolAcknowledged(ctx sdk.Context, msg *types.MsgCreatePoolRequest) error {
 	//save pool after complete create operation in counter party chain.
 
 	pool := types.NewInterchainLiquidityPool(
 		ctx,
+		msg.Sender,
 		k.bankKeeper,
-		msg.Denoms,
+		msg.Tokens,
 		msg.Decimals,
 		msg.Weight,
 		msg.SourcePort,
 		msg.SourceChannel,
 	)
 
+	// when add initial liquidity, we need to update pool token amount.
+	initialLiquidity := sdk.NewCoin(pool.PoolId, msg.Tokens[0].Amount)
+	pool.AddPoolSupply(initialLiquidity)
+
+	// mint lpToken.
+	err := k.MintTokens(ctx, sdk.MustAccAddressFromBech32(msg.Sender), initialLiquidity)
```

Comment:
it would be risky since the counterparty deposit is not finished.

Reply:
hmm, if counterparty deposit is not finished, that lp token has no meaning forever. so I think it's no risk. 
Only when deposit is finished in counter party side, at that time pool is ready for swap and lp token has meaning.

Reply:
and then if counter party deposit is not finished and pool creator has no ability to treat that, at that time, he can withdraw his fund with that ip token from pool. 

of course at that time, we need to delete pool also, about that operation, need to improve.